### PR TITLE
Feature/vault integration

### PR DESCRIPTION
**Pull request by alpaca4j**

### CODE REVIEW

File: src/main/scala/com/ing/wbaa/rokku/sts/config/VaultSettings.scala [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+import akka.actor.{ ExtendedActorSystem, Extension, ExtensionId, ExtensionIdProvider }
+import com.typesafe.config.Config
+
+class VaultSettings(config: Config) extends Extension {
+
+  val vaultUrl: String = config.getString("vault.url")
+  val vaultPath: String = config.getString("vault.path")
+  val vaultToken: String = config.getString("vault.token")
+  val readTimeout: Int = config.getString("vault.read-timeout").toInt
```
Comment (arempter) [inline src/main/scala/com/ing/wbaa/rokku/sts/config/VaultSettings.scala:11]: I think you can use getInt directly, like conf.getInt("foo.bar")

File: src/main/scala/com/ing/wbaa/rokku/sts/vault/VaultService.scala [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+import scala.jdk.CollectionConverters._
+import scala.util.{Failure, Success, Try}
+
+trait VaultService extends LazyLogging {
+
+  protected[this] def vaultSettings: VaultSettings
+
+  protected[this] implicit def system: ActorSystem
+
+  protected[this] implicit lazy val vaultExecutionContext: ExecutionContext =
```
Comment (arempter) [inline src/main/scala/com/ing/wbaa/rokku/sts/vault/VaultService.scala:23]: like above just use one we already have

File: src/main/scala/com/ing/wbaa/rokku/sts/vault/VaultService.scala [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+      .engineVersion(2)
+      .token(vaultSettings.vaultToken)
+      .readTimeout(vaultSettings.readTimeout)
+      .openTimeout(vaultSettings.openTimeout)
+      .build())
+    vault
+  }
+
+  def insertNpaCredentialsToVault(username: UserName, awsCredential: AwsCredential): Future[Boolean] = Future {
+    val secretsToSave: Map[String, AnyRef] = Map("accessKey" -> awsCredential.accessKey.value, "secretKey" -> awsCredential.secretKey.value)
```
Comment (arempter) [inline src/main/scala/com/ing/wbaa/rokku/sts/vault/VaultService.scala:45]: what is the reason for AnyRef for secretKey? I think it should be at least string as well?

**Review by arempter — CHANGES_REQUESTED** [review]

File: src/main/scala/com/ing/wbaa/rokku/sts/vault/VaultService.scala (hunk 3 above)
Comment (alpaca4j) [reply src/main/scala/com/ing/wbaa/rokku/sts/vault/VaultService.scala:45]: Vault client dependency method signature for writing is as follows:
`public LogicalResponse write(final String path, final Map<String, Object> nameValuePairs)` and with no overloads which provide a different String generic.

The vaule is being used here:

`.write(vaultSettings.vaultPath + "/" + username.value, secretsToSave.asJava)`

However `insertNpaCredentialsToVault` takes `UserName` and `AwsCredential` objects as parameters, hence I don't see this as a potential problem.

File: src/main/resources/application.conf [hunk 4]
```diff
@@ -56,3 +56,23 @@ db-dispatcher {
...
+vault {
+    url = ${?VAULT_URL}
+    path = ${?VAULT_SECRET_PATH}
+    token = ${?VAULT_TOKEN}
+    retries = ${?VAULT_RETRIES}
+    read-timeout= ${?VAULT_READ_TIMEOUT}
+    open-timeout = ${?VAULT_OPEN_TIMEOUT}
+}
+
+vault-dispatcher {
```
[the remark this answers is not in the document]
Comment (alpaca4j) [reply src/main/resources/application.conf]: fixed

File: src/main/scala/com/ing/wbaa/rokku/sts/config/VaultSettings.scala (hunk 1 above)
Comment (alpaca4j) [reply src/main/scala/com/ing/wbaa/rokku/sts/config/VaultSettings.scala]: fixed

File: src/main/scala/com/ing/wbaa/rokku/sts/vault/VaultService.scala (hunk 2 above)
Comment (alpaca4j) [reply src/main/scala/com/ing/wbaa/rokku/sts/vault/VaultService.scala]: fixed

**Review by arempter — APPROVED** [review]

**Pull request merged**